addons/account/tests/test_account_journal_dashboard.py

```diff
@@ -100,36 +104,123 @@ def test_sale_purchase_journal_for_multi_currency_purchase(self):
         currency = self.currency_data['currency']
         company_currency = self.company_data['currency']
 
-        invoice = self.env['account.move'].create({
-            'move_type': 'in_invoice',
-            'invoice_date': '2017-01-01',
-            'date': '2017-01-01',
-            'partner_id': self.partner_a.id,
-            'currency_id': currency.id,
-            'invoice_line_ids': [
-                (0, 0, {'name': 'test', 'price_unit': 200})
-            ],
-        })
-        invoice.action_post()
+        invoice = self.init_invoice('in_invoice', invoice_date='2017-01-01', post=True, amounts=[200], currency=currency)
+        _draft_invoice = self.init_invoice('in_invoice', invoice_date='2017-01-01', post=False, amounts=[200], currency=currency)
 
-        payment = self.env['account.payment'].create({
-            'amount': 90.0,
-            'date': '2016-01-01',
-            'payment_type': 'outbound',
-            'partner_type': 'supplier',
-            'partner_id': self.partner_a.id,
-            'currency_id': currency.id,
-        })
-        payment.action_post()
+        payment = self.init_payment(-90, post=True, date='2017-01-01', currency=currency)
 
         (invoice + payment.move_id).line_ids.filtered_domain([
             ('account_id', '=', self.company_data['default_account_payable'].id)
         ]).reconcile()
 
         default_journal_purchase = self.company_data['default_journal_purchase']
-        dashboard_data = default_journal_purchase._get_journal_dashboard_data_batched()[default_journal_purchase.id]
-        self.assertEqual(format_amount(self.env, 55, company_currency), dashboard_data['sum_waiting'])
-        self.assertEqual(format_amount(self.env, 55, company_currency), dashboard_data['sum_late'])
+        expected_values = {
+            'number_draft': 1,
+            'sum_draft': format_amount(self.env, 100, company_currency),
+            'number_waiting': 1,
+            'sum_waiting': format_amount(self.env, 55, company_currency),
+            'number_late': 1,
+            'sum_late': format_amount(self.env, 55, company_currency),
+        }
+        self.assertDashboardData(default_journal_purchase, expected_values)
+
+    def test_sale_purchase_journal_foreign_currency_for_purchase(self):
```

Comment:
The only different thing from this function and `test_sale_purchase_journal_company_currency_for_purchase` is the currency. Maybe we can combine them in 1 function, and just do a loop instead?

Maybe the same can also be done for the others? (`test_sale_purchase_journal_foreign_currency_for_purchase_company_currency` and `test_sale_purchase_journal_company_currency_for_purchase_foreign_currency`)